tests/integration/test_renewals.cairo

```diff
@@ -0,0 +1,242 @@
+        warp(5180000, context.renewal_contract)
+        stop_mock = mock_call(123, "transferFrom", [1])
+    %}
+
+    // Test if the automatic renewal has been toggled for this domain
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (renew) = Renewal.will_renew(renewal_contract, TH0RGAL_STRING, 456);
+    assert renew = 1;
+
+    %{ stop_prank_callable() %}
+
+    // Test renew domain for 1 year
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+    assert_expiry(TH0RGAL_STRING, (60 * 86400) + (365 * 86400) + 1);
+
+    %{ stop_mock() %}
+
+    return ();
+}
+
+@external
+func test_renew_expired_domain{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    buy_domain(60);
+    tempvar renewal_contract;
+    tempvar naming;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        ids.naming = context.naming_contract
+        stop_prank_callable = start_prank(456, target_contract_address=ids.renewal_contract)
+        warp(10000000, context.renewal_contract)
+        stop_mock = mock_call(123, "transferFrom", [1])
+    %}
+
+    // Test if the automatic renewal has been toggled for this domain
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (renew) = Renewal.will_renew(renewal_contract, TH0RGAL_STRING, 456);
+    assert renew = 1;
+
+    %{ stop_prank_callable() %}
+
+    // Test renew domain for 1 year even though domain was expired
+    Renewal.renew(renewal_contract, TH0RGAL_STRING, 456);
+    assert_expiry(TH0RGAL_STRING, (60 * 86400) + (365 * 86400) + 1);
+
+    %{ stop_mock() %}
+
+    return ();
+}
+
+@external
+func test_renew_domains{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
+    alloc_locals;
+    buy_domain(60);
+    tempvar renewal_contract;
+    tempvar naming;
+    %{
+        ids.renewal_contract = context.renewal_contract
+        ids.naming = context.naming_contract
+        stop_prank_callable = start_prank(456, target_contract_address=ids.renewal_contract)
+        warp(5180000, context.renewal_contract)
+        stop_mock = mock_call(123, "transferFrom", [1])
+    %}
+
+    // Test if the automatic renewal has been toggled for both domains
+    Renewal.toggle_renewals(renewal_contract, TH0RGAL_STRING);
+    let (renew) = Renewal.will_renew(renewal_contract, TH0RGAL_STRING, 456);
+    assert renew = 1;
+
+    Renewal.toggle_renewals(renewal_contract, ANOTHER_DOMAIN);
+    let (renew_2) = Renewal.will_renew(renewal_contract, ANOTHER_DOMAIN, 456);
+    assert renew_2 = 1;
+
+    %{ stop_prank_callable() %}
+
+    // Test renewing both domains for 1 year
+    Renewal.batch_renew(renewal_contract, 2, cast(new(TH0RGAL_STRING, ANOTHER_DOMAIN), felt*), 2, cast(new(456, 456), felt*));
+
+    assert_expiry(TH0RGAL_STRING, (60 * 86400) + (365 * 86400) + 1);
+    assert_expiry(ANOTHER_DOMAIN, (60 * 86400) + (365 * 86400) + 1);
+
+    %{ stop_mock() %}
+
+    return ();
+}
```

Comment:
Did you test when the domain is already 1 year expiry (should fail) ? 

Did you test to renew for more than one year (should fail) ?